scripts/builder/checker_builder/test/test_config.py

```diff
@@ -1,6 +1,88 @@
+from io import StringIO
+
 import pytest
+import yaml
 from marshmallow import ValidationError
 
+DECIMAL_YAML = """
+some_int: 1
+some_str: foo
+some_ratio: 1/42
+"""
+
+
+def test_serialize_ratio():
+    from checker_builder.config import Ratio, RatioField
+
+    input = Ratio(1, 3)
+    assert RatioField().serialize("ratio", obj={"ratio": input}) == "1/3"
+
+
+def test_deserialize_ratio():
+    from checker_builder.config import Ratio, RatioField
+
+    assert RatioField().deserialize("1/9") == Ratio(1, 9)
+
+
+def test_deserialize_ratio_zero_den_fails():
+    from checker_builder.config import RatioField
+
+    with pytest.raises(ValidationError):
+        RatioField().deserialize("1/0")
+
+
+def test_serialiation_roundtrip_ratio():
+    from checker_builder.config import Ratio, RatioField
+
+    input = Ratio(-3, 997)
+    field = RatioField()
+    assert field.deserialize(field.serialize("ratio", obj={"ratio": input})) == input
+
+
+def test_deserialize_positive_ratio():
+    from checker_builder.config import PositiveRatioField, Ratio
+
+    assert PositiveRatioField().deserialize("-1/-9") == Ratio(-1, -9)
```

Comment:
Hmmm, actually now that I look at this, we don't want to allow ratios with a negative denominator (i.e., this will fail on the OCaml side). In general we should have `den > 0` and get the sign from the numerator.